src/components/director-view/director-view.jsx

```diff
@@ -1,33 +1,80 @@
-import React from "react";
-import { useSelector } from "react-redux";
-import { useParams } from "react-router";
+import React, { useState, useEffect } from "react";
+import { Container, Row, Col } from "react-bootstrap";
+import { useParams, useNavigate } from "react-router-dom";
+import PropTypes from "prop-types";
 
+export const DirectorView = ({ token }) => {
+  const { directorName } = useParams();
+  const navigate = useNavigate();
+  const [director, setDirector] = useState(null);
 
-export const DirectorView = () => {
-  const params = useParams();
-  const directors = useSelector((state) => state.directors.list);
-  const director = directors.find((director) => director.name === params.name);
-
+  useEffect(() => {
+    if (!token) return;
+    (async () => {
+      try {
+        const response = await fetch(
+          `https://movieapi-aeueoes-projects.vercel.app/directors/${directorName}`,
+          {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          }
+        );
+        const data = await response.json();
+        setDirector(data);
+      } catch (error) {
+        console.error("Error fetching director:", error);
+      }
+    })();
+  }, [token, directorName]);
   if (!director) {
-    return <div>Director not found!</div>;
+    return <div>Loading...</div>; // or a loading spinner, etc.
```

Comment:
Is it actually loading of not found. the negation symbol ! seems to me like the item should be not found instead